examples/llil_parser/src/LowLevel_IL_Parser.cpp

```diff
@@ -0,0 +1,433 @@
+			m_tabs += 3;
+		}
+		else if (operand.type == OperandType::kInt)
+		{
+			// In this case the operand is simply a value
+			showIndent();
+			printf("Value: %zX\n", insn.operands[0]);
+			m_tabs += 3;
+		}
+		else if (operand.type == OperandType::kFlag)
+		{
+			// In this case the operand is a flag
+			printf("Flag: %s\n", m_bv->GetDefaultArchitecture()->GetFlagName(static_cast<uint32_t>(insn.operands[0])).c_str());
+			m_tabs += 3;
+		}
+		else if (operand.type == OperandType::kIntList)
+		{
+			// In this case we have an array of llil targets
+			std::vector<uint64_t> intList = llil->GetOperandList(llil->GetIndexForInstruction(m_currentInstructionId), operandId);
+			showIndent();
+			printf("Target LLIL Indices: ");
+			for (const auto i : intList)
+			{
+				printf("%zd ", i);
+			}
+			printf("\n");
+		}
+		else
+		{
+			printf("[e] LLIL Parser: Not Handled -> OperandPurpose: %d OperandType: %d\n", operand.purpose, operand.type);
+		}
+
+
+		operandId++;
+	}
+
+
+}
+
+void LlilParser::decodeIndexInFunction(uint64_t functionAddress, int indexIl)
+{
+
+	m_currentFunction = m_bv->GetAnalysisFunctionsForAddress(functionAddress);
+	if (m_currentFunction.size() < 1)
+		throw std::runtime_error("Error no functions at requested address\n");
+
+	BinaryNinja::Function *function = m_currentFunction[0];
+	BinaryNinja::Ref<BinaryNinja::LowLevelILFunction> llil = function->GetLowLevelIL();
+
+	m_currentInstructionId = indexIl;
+	BNLowLevelILInstruction currentInstruction = llil->operator[](llil->GetIndexForInstruction(indexIl));
+
+
+	analysisInstruction(currentInstruction);
+	m_tabs = 0;
+
+}
+
+void LlilParser::decodeWholeFunction(BinaryNinja::Function *function)
+{
+	m_currentFunction.clear();
+	m_currentFunction.push_back(function);
+
+	BinaryNinja::Ref<BinaryNinja::LowLevelILFunction> llil = function->GetLowLevelIL();
+
+	for (size_t i = 0; i < llil->GetInstructionCount(); i++)
+	{
+		
+		m_currentInstructionId = i;
+		BNLowLevelILInstruction currentInstruction = llil->operator[](llil->GetIndexForInstruction(i));
+		
+		printf("\n[%zx][%zd]---------------------------------------------------------------------------\n", currentInstruction.address, i);
+
+		analysisInstruction(currentInstruction);
+		m_tabs = 0;
+	}
+
+}
+
+void LlilParser::decodeWholeFunction(uint64_t functionAddress)
+{
+
+	m_currentFunction = m_bv->GetAnalysisFunctionsForAddress(functionAddress);
+	if (m_currentFunction.size() < 1)
+		throw std::runtime_error("Error no functions at requested address or possible invalid BundledPluginDirectory\n");
+
+	BinaryNinja::Function *function = m_currentFunction[0];
+	BinaryNinja::Ref<BinaryNinja::LowLevelILFunction> llil = function->GetLowLevelIL();
+
+	
+
+	for (size_t i = 0; i < llil->GetInstructionCount(); i++)
+	{
+		m_currentInstructionId = i;
+		BNLowLevelILInstruction currentInstruction = llil->operator[](llil->GetIndexForInstruction(i));
```

Comment:
Perhaps a little cleaner than the above:
`BNLowLevelILInstruction currentInstruction = (*llil)[llil->GetIndexForInstruction(i)];`